packages/@expo/fingerprint/src/__tests__/ProjectWorkflow-test.ts

```diff
@@ -0,0 +1,103 @@
+
+const defaultSpawnAsync = async (command, args) => {
+  if (command === 'git' && args[0] === '--help') {
+    return { status: 0 };
+  }
+  if (command === 'git' && args[0] === 'rev-parse' && args[1] === '--show-toplevel') {
+    return { stdout: '/app' };
+  }
+  throw new Error(`Unexpected command: ${command} ${args.join(' ')}`);
+};
+jest.mock('@expo/spawn-async', () => ({
+  __esModule: true,
+  default: jest.fn().mockImplementation(defaultSpawnAsync),
+}));
+jest.mock('fs');
+jest.mock('fs/promises');
+
+describe(resolveProjectWorkflowAsync, () => {
+  afterEach(() => {
+    vol.reset();
+  });
+
+  it('should return managed workflow for a empty project', async () => {
+    expect(await resolveProjectWorkflowAsync('/app', 'android', [])).toEqual('managed');
+    expect(await resolveProjectWorkflowAsync('/app', 'ios', [])).toEqual('managed');
+  });
+
+  it('should return generic workflow for a project with a build.gradle file', async () => {
+    vol.fromJSON({
+      '/app/android/app/build.gradle': '',
+    });
+    expect(await resolveProjectWorkflowAsync('/app', 'android', [])).toEqual('generic');
+  });
+
+  it('should return generic workflow for a project with a pbxproj file', async () => {
+    vol.fromJSON({
+      '/app/ios/app.xcodeproj/project.pbxproj': '',
+    });
+    expect(await resolveProjectWorkflowAsync('/app', 'ios', [])).toEqual('generic');
+  });
+
+  it('should return managed workflow for a project with native project files and ignored by .fingerprintignore', async () => {
+    const ignorePaths = buildPathMatchObjects(['android/**/*', 'ios/**/*']);
+    vol.fromJSON({
+      '/app/android/app/build.gradle': '',
+      '/app/ios/app.xcodeproj/project.pbxproj': '',
+    });
+    expect(await resolveProjectWorkflowAsync('/app', 'android', ignorePaths)).toEqual('managed');
+    expect(await resolveProjectWorkflowAsync('/app', 'ios', ignorePaths)).toEqual('managed');
+  });
+
+  it('should return managed workflow for a project with native project files and ignored by gitignore', async () => {
+    const mockSpawnAsync = spawnAsync as jest.MockedFunction<typeof spawnAsync>;
+    mockSpawnAsync.mockImplementation(async (command, args, options) => {
+      if (command === 'git' && args[0] === 'check-ignore' && args[1] === '-q') {
+        if (['android/app/build.gradle', 'ios/app.xcodeproj/project.pbxproj'].includes(args[2])) {
+          return { status: 0 };
+        } else {
+          throw new Error('Not gitignored');
+        }
+      }
+      return defaultSpawnAsync(command, args);
+    });
+
+    vol.fromJSON({
+      '/app/android/app/build.gradle': '',
+      '/app/ios/app.xcodeproj/project.pbxproj': '',
+    });
+    expect(await resolveProjectWorkflowAsync('/app', 'android', [])).toEqual('managed');
+    expect(await resolveProjectWorkflowAsync('/app', 'ios', [])).toEqual('managed');
+
+    mockSpawnAsync.mockReset();
+  });
+});
+
+describe('config-plugins API', () => {
```

Comment:
Let's also put a copy of this test in the config-plugins package so that changes to that package trigger a test failure if the API is changed.

Reply:
interesting! just know that we don't have tests for these functions.
added in the pr https://app.graphite.dev/github/pr/expo/expo/37287/%5Bconfig-plugins%5D-add-unit-test-for-Paths